bundle/regal/config/exclusion_test.rego

```diff
@@ -0,0 +1,79 @@
+package regal.config_test
+
+import data.regal.config
+
+# map[pattern: string]map[filename: string]excluded: bool
+cases := {
+	"p.rego": {
+		"p.rego": true,
+		"dir/p.rego": true,
+		"dir1/dir2/p.rego": true,
+		"q.rego": false,
+	},
+	"**/p.rego": {
+		"p.rego": true,
+		"dir/p.rego": true,
+		"dir1/dir2/p.rego": true,
+		"q.rego": false,
+	},
+	"dir/**/p.rego": {
+		"dir/sub/p.rego": true,
+		"dir2/sub/p.rego": false,
+		"dir/p.rego": true,
+		"dir2/p.rego": false,
+		"dir/sub/sub1/p.rego": true,
+		"dir/sub/q.rego": false,
+	},
+	"*_test.rego": {
+		"p_test.rego": true,
+		"q_test.rego": true,
+		"dir/p_test.rego": true,
+		"p.rego": false,
+		"ptest.rego": false,
+	},
+}
+
+test_all_cases_are_as_expected {
+	not_exp := {pattern: res |
+		subcases := cases[pattern]
+		res := {file: res1 |
+			exp := subcases[file]
+			act := config.exclude(pattern, file)
+			exp != act
+			res1 := {"exp": exp, "act": act}
+		}
+		count(res) > 0
+	}
+
+	count(not_exp) == 0
+}
+
+rule_specific_config := {"ignore": ["p.rego"]}
```

Comment:
Is this really rule specific? I don't see where it's attached to some specific rule.

Reply:
I believe it's memoized given the same args, but honestly not sure ... we should probably add some docs on that in OPA 😅 @srenatus would know, I think.

What I meant by a rule specific ignore would be something like:

```yaml
rules:
  testing:
    print-or-trace-call:
      level: error
      ignore:
        files:
          - **/*_test.rego
```

Reply:
Or perhaps I'm just too tired and this is already how things work 😅 All good if so! But a test case that made this more clear would be great.

Reply:
As discussed on Slack, ignoring files is possible at the rule level, global config level and through the CLI flag `ignore-files`.

Also, the interface of the `exclusion.rego` file is made similar to that of the `config.for_rule`, so it now takes `metadata` as input and gets the config (presumably memoized, but likely not a big factor for performance anyway). 

I have added e2e tests, unit tests and more clear policy tests now. 

Can you have another look when you have time and let me know what you think.